olmo/train.py

```diff
@@ -711,6 +711,12 @@ def train_step(self, batch: Dict[str, Any], reduce_global_loss: bool = True) ->
             process_group=self.fsdp_model.process_group,
         )
 
+        # TODO: what to do otherwise?
+        if should_log_optim_metrics_this_step:
+            emb_decay_factor = 1.0 - optim_metrics["param/transformer.wte.weight.norm"]
+        else:
+            emb_decay_factor = 1.0
```

Comment:
Norm is not the same as stddev. `sqrt(norm^2 / (embedding_size * vocab size))` is the stddev.

We produce this value every step, because we need it for grad norm. We just don't output it every step. Can you change it so we output it every step, and then we don't have a special case here?

Reply:
We compute the norm of the gradient every step (`grad/transformer.wte.weight.norm`), not the norm of the parameter itself (`param/transformer.wte.weight.norm`). Don't we need the latter?